pkg/controller/controller_binding.go

```diff
@@ -351,6 +351,13 @@ func (c *controller) reconcileServiceBindingDelete(binding *v1beta1.ServiceBindi
 
 	instance, err := c.instanceLister.ServiceInstances(binding.Namespace).Get(binding.Spec.ServiceInstanceRef.Name)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			glog.V(5).Info(pcb.Messagef(`References a non-existent %s "%s/%s"`,
+				pretty.ServiceInstance, binding.Namespace, binding.Spec.ServiceInstanceRef.Name,
+			))
+			return c.processUnbindSuccess(binding)
```

Comment:
I am worried that this allows a binding to be lost if the class changes underneath the instance type temporarily. 

In the very least this check needs to look at `binding.Status.UnbindStatus` to see if there had been a successful binding request sent to the broker before we delete this from kubernetes because otherwise this will leak a binding on the broker side even if it happens to be in a bad state at the moment.

Reply:
According to the OSB, classes can't change. You can have a new class that's offered, but it's never related to the old one.

So if there is no reference to an instance or class, we will never be able to issue a meaningful request to the broker to delete the binding. If it's in a bad state now, how is it ever going to not be in a bad state in the future?

Reply:
I think we should throw an error instead and keep the binding in the "broken deletion" state instead of silently proceeding.